### PR TITLE
Adding support for resetting the redux-store when the component is unmounted

### PR DESCRIPTION
**Pull request by Dananji**

### CODE REVIEW

**Comment by adamjarling** [comment]
Hey @Dananji this looks good to me, and all seems to work ok now?  I can't test it locally.  Everything seems to work with multiple occurrences?

File: src/reducers/index.js [hunk 1]
```diff
@@ -2,9 +2,20 @@ import { combineReducers } from 'redux';
...
+const appReducer = combineReducers({
   forms,
   smData,
   peaksInstance
 });
+
+const rootReducer = (state, action) => {
+  if (action.type === types.RESET_STORE) {
+    // Reducers return initial state when they are called with 'undefined'
+    state = undefined;
```
Comment (adamjarling) [inline src/reducers/index.js]: Yea, my only question is if this should be `state = {}` instead of an `undefined` value.

**Comment by Dananji** [comment]
@adamjarling Yeah, I ingested multiple items with more than one sections, so we will have more than one instance of SME. And it all works. The state is reset to initial state whenever the modal containing SME is closed.

**Review by adamjarling — APPROVED** [review]

File: src/reducers/index.js (hunk 1 above)
Comment (Dananji) [reply src/reducers/index.js]: The reducers set the state to initial state only when state is set to `undefined` (not declared/declared but has no value) not any other value. So the way to reset the state to its initial value is to pass `state = undefined` into the reducers.

**Pull request merged**